src/golang/lib/workflow/orchestrator/orchestrator.go

```diff
@@ -29,479 +19,201 @@ const (
 )
 
 var (
-	ErrIncorrectOperatorsScheduled = errors.New("Incorrect number of operators scheduled.")
-	ErrOpExecSystemFailure         = errors.New("Operator execution failed due to system error.")
-	ErrOpExecBlockingUserFailure   = errors.New("Operator execution failed due to user error.")
-	ErrInvalidOpId                 = errors.New("Invalid operator ID.")
-	ErrInvalidArtifactId           = errors.New("Invalid artifact ID.")
+	ErrOpExecSystemFailure       = errors.New("Operator execution failed due to system error.")
+	ErrOpExecBlockingUserFailure = errors.New("Operator execution failed due to user error.")
 )
 
-func initializeOrchestration(
-	operators map[uuid.UUID]operator.Operator,
-	ready map[uuid.UUID]bool,
-	operatorDependencies map[uuid.UUID]map[uuid.UUID]bool,
-	artifactToDownstreamOperatorIds map[uuid.UUID][]uuid.UUID,
-) {
-	// This block does an initial scan of all operators and set the initial state for the orchestration.
-	// Input data structures should be empty and will be updated throughout initialization:
-	// - set upstream artifact counts for all operators
-	// - put all operators without any upstream to 'ready to schedule' set
-	// - set downstream operator list for all artifacts
-	for id, operator := range operators {
-		operatorDependencies[id] = make(map[uuid.UUID]bool, len(operator.Inputs))
-		for _, artifactId := range operator.Inputs {
-			operatorDependencies[id][artifactId] = true
-		}
+type Orchestrator interface {
+	Execute(ctx context.Context, dag dag.WorkflowDag) (shared.ExecutionStatus, error)
+}
 
-		if len(operator.Inputs) == 0 {
-			ready[id] = true
-		}
+type orchestratorImpl struct {
```

Comment:
How aboue 'aqueductOrchestrator' to be consistent with how we name server, and also we know we will have other 'impl' of orchestrators